hack/generate-changelog.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o pipefail
+this=`basename $0`
+
+usage () {
+cat << EOF
+Generate a changelog for the specified tag
+Usage: $this --reference <tag> [--remote <remote_name>]
+
+Options:
+  --reference specify the tag to start the changelog from (default: latest tag)
+  --remote    specify the remote to fetch tags from (default: origin)
+  --version   specify the version to be released
+  --help/-h   show this help and exit
+
+EOF
+}
+
+# Fetch the latest tags from the specified remote
+git fetch "$REMOTE" --tags
+
+# Define default starting tag
+REFERENCE=$(git describe --tags $(git rev-list --tags --max-count=1))
+REMOTE="origin"
+VERSION=""
+
+# Parse command line options
+while [[ $# -gt 0 ]]; do
+    key="$1"
+    case $key in
+        --reference)
+        REFERENCE="$2"
+        shift # past argument
+        shift # past value
+        ;;
+        --remote)
+        REMOTE="$2"
+        shift # past argument
+        shift # past value
+        ;;
+        --version)
+        VERSION="$2"
+        shift # past argument
+        shift # past value
+        ;;
+        --help/-h)  usage
+            exit 0
+            ;;
+        *)  usage
+            exit 1
+            ;;
+    esac
+done
+
+# Define temporary file for storing raw commit messages
+TEMP_FILE=$(mktemp)
+
+# Define function to print formatted section header
+print_section_header() {
+    echo "## $1"
+    echo
+    echo "### Version: $REFERENCE"
+}
+
+# Get the range of commits from the starting tag to the current commit
+CHANGELOG=$(git log --pretty=format:"%s" $latest_tag..@)
+
+# Output the changes into the temporary file
+echo "$CHANGELOG" > "$TEMP_FILE"
```

Comment:
Why not just have the git command here directly? Do we really need to have it defined as a variable?